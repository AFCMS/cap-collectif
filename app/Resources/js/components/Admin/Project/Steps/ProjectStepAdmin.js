// @flow
import React, { useState } from 'react';
import { FieldArray } from 'redux-form';
import { Button, ButtonToolbar } from 'react-bootstrap';
import { FormattedMessage } from 'react-intl';

import ProjectStepAdminList from './ProjectStepAdminList';
import ProjectAdminStepFormModal from '../Form/ProjectAdminStepFormModal';

type Props = {
  form: string,
};

export default function ProjectStepAdmin(props: Props) {
  const [showAddStepModal, displayAddStepModal] = useState(false);

  const { form } = props;

  return (
    <div className="col-md-12">
      <div className="box box-primary container-fluid">
        <div className="box-header">
          <h4 className="box-title">
            <FormattedMessage id="admin.group.steps" />
          </h4>
        </div>
        <div className="box-content">
          <div className="form-group" id="proposal_form_admin_questions_panel_personal">
            <FieldArray name="steps" component={ProjectStepAdminList} formName={form} />
            <ButtonToolbar>
              <ProjectAdminStepFormModal
                onClose={() => displayAddStepModal(false)}
                step={null}
                show={showAddStepModal}
                form={form}
              />
              <Button
                id="js-btn-create-question"
                bsStyle="primary"
                className="btn-outline-primary box-content__toolbar"
                onClick={() => displayAddStepModal(true)}>
                <i className="cap cap-bubble-add-2" />{' '}
                <FormattedMessage id="question_modal.create.title" />
              </Button>
            </ButtonToolbar>
          </div>
        </div>
      </div>
    </div>
  );
}
