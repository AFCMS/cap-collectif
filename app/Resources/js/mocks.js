// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
