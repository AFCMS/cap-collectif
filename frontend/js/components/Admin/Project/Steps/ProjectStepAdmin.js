// @flow
import React, { useState } from 'react';
import { FieldArray } from 'redux-form';
import { createFragmentContainer, graphql } from 'react-relay';
import { ButtonToolbar } from 'react-bootstrap';
import { connect } from 'react-redux';
import { FormattedMessage, type IntlShape } from 'react-intl';
import ProjectStepAdminList from './ProjectStepAdminList';
import ProjectAdminStepFormModal from '../Step/ProjectAdminStepFormModal';
import {
  ProjectBoxHeader,
  StepMenuItem,
  ProjectBoxContainer,
  StepCreateButton,
} from '../Form/ProjectAdminForm.style';
import { STEP_TYPES } from '~/constants/StepTypeConstants';
import { type StepTypes } from '../Form/ProjectAdminForm';
import type { ProjectStepAdmin_project } from '~relay/ProjectStepAdmin_project.graphql';
import type { GlobalState } from '~/types';

type Props = {|
  ...ReduxFormFormProps,
  form: string,
  intl: IntlShape,
  project: ProjectStepAdmin_project,
  hasFeatureDebate: boolean,
|};

export const validate = ({ steps }: StepTypes) => {
  const errors = {};

  /* AbstractStep */
  const titles = steps.map(s => s.title.toLowerCase());

  if (titles.some((item, index) => titles.indexOf(item) !== index)) {
    errors.steps = { _error: 'unique-step-title' };
  }

  /* QuestionnaireStep */
  const questionnaires = steps
    .filter(s => s.type === 'QuestionnaireStep')
    .map(s => s.questionnaire.value);

  if (questionnaires.some((item, index) => questionnaires.indexOf(item) !== index)) {
    errors.steps = { _error: 'duplicate-questionnaire' };
  }

  return errors;
};

export const ProjectStepAdmin = ({ form, project, hasFeatureDebate }: Props) => {
  const [stepType, setStepType] = useState('OtherStep');
  const [showAddStepModal, displayAddStepModal] = useState(false);

  const stepTypes = React.useMemo(() => {
    if (!hasFeatureDebate) {
      return STEP_TYPES.filter(step => step.value !== 'DebateStep');
    }

    return STEP_TYPES;
  }, [hasFeatureDebate]);

  return (
    <div className="col-md-12">
      <ProjectBoxContainer className="box container-fluid">
        <ProjectBoxHeader>
          <h4>
            <FormattedMessage id="project.show.meta.step.title" />
          </h4>
        </ProjectBoxHeader>
        <div className="box-content">
          <div className="form-group" id="project_form_admin_steps_panel">
            <FieldArray
              name="steps"
              component={ProjectStepAdminList}
              formName={form}
              project={project}
            />
            <ButtonToolbar>
              <ProjectAdminStepFormModal
                onClose={() => displayAddStepModal(false)}
                step={null}
                isCreating
                type={stepType}
                show={showAddStepModal}
                form={form}
                project={project}
              />
              <StepCreateButton
                id="js-btn-create-step"
                bsStyle="primary"
                title={
                  <>
                    <i className="fa fa-plus-circle" /> <FormattedMessage id="global.add" />
                  </>
                }>
                {stepTypes.map((st, idx) => (
                  <StepMenuItem
                    key={idx}
                    id={st.label}
                    onClick={() => {
                      setStepType(st.value);
                      displayAddStepModal(true);
                    }}>
                    <FormattedMessage id={st.label} />
                  </StepMenuItem>
                ))}
              </StepCreateButton>
            </ButtonToolbar>
          </div>
        </div>
      </ProjectBoxContainer>
    </div>
  );
};

const mapStateToProps = (state: GlobalState) => ({
  hasFeatureDebate: state.default.features.unstable__debate || false,
});

const ProjectStepAdminConnected = connect(mapStateToProps)(ProjectStepAdmin);

export default createFragmentContainer(ProjectStepAdminConnected, {
  project: graphql`
    fragment ProjectStepAdmin_project on Project {
      ...ProjectAdminStepFormModal_project
      ...ProjectStepAdminList_project
    }
  `,
});
