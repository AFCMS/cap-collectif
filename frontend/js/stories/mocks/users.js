// @flow

export const author = {
  id: 'user1',
  username: 'admin',
  media: {
    url: 'https://source.unsplash.com/random/150x150',
  },
  _links: {},
  vip: false,
  isViewer: false,
  isAdmin: true,
  isEvaluer: true,
};

export const followers = [
  {
    displayName: 'All Fred',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
  {
    displayName: 'Louisette',
    isEmailConfirmed: false,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 1,
  },
  {
    contributionsCount: 1,
    isEmailConfirmed: true,
    _links: null,
    media: null,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
];
