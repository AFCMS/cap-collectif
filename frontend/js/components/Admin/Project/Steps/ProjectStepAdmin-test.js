// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ProjectStepAdmin } from './ProjectStepAdmin';
import { $fragmentRefs, $refType, formMock, intlMock } from '~/mocks';

const baseProps = {
  ...formMock,
  intl: intlMock,
  form: 'testForm',
  project: {
    $fragmentRefs,
    $refType,
  },
  query: {
    $fragmentRefs,
    $refType,
  },
  viewerIsAdmin: true,
  hasIdentificationCodeLists: true,
};

const props = {
  basic: baseProps,
};

describe('<ProjectStepAdmin />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ProjectStepAdmin {...props.basic} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with isAdmin false', () => {
    const wrapper = shallow(<ProjectStepAdmin {...props.basic} viewerIsAdmin={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
