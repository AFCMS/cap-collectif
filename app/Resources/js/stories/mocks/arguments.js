// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
