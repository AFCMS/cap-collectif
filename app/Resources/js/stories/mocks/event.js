// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
