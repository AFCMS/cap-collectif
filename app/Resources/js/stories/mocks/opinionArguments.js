// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
