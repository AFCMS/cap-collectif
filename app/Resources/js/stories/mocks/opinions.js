// @flow
import { author } from './users';

export const opinion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  createdAt: ' • 1 mars 2018',
  pinned: true,
  publishedAt: ' • 1 mars 2018',
  votes: { totalCount: 4 },
  versions: { totalCount: 3 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, votes: { totalCount: 0 } },
];
