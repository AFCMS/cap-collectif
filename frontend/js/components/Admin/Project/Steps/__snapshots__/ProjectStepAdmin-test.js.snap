// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectStepAdmin /> renders correctly 1`] = `
<div
  className="col-md-12"
>
  <ProjectAdminFormstyle__ProjectBoxContainer
    className="box container-fluid"
  >
    <ProjectAdminFormstyle__ProjectBoxHeader>
      <h4>
        <FormattedMessage
          id="project.show.meta.step.title"
          values={Object {}}
        />
      </h4>
    </ProjectAdminFormstyle__ProjectBoxHeader>
    <div
      className="box-content"
    >
      <div
        className="form-group"
        id="project_form_admin_steps_panel"
      >
        <FieldArray
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "__ComponentClass": [Function],
              "displayName": "Relay(Connect(ProjectStepAdminList))",
              "render": [Function],
            }
          }
          formName="testForm"
          name="steps"
          project={
            Object {
              "$fragmentRefs": null,
              "$refType": null,
            }
          }
        />
        <ButtonToolbar
          bsClass="btn-toolbar"
        >
          <Relay(ProjectAdminStepFormModal)
            form="testForm"
            isCreating={true}
            onClose={[Function]}
            project={
              Object {
                "$fragmentRefs": null,
                "$refType": null,
              }
            }
            show={false}
            step={null}
            type="OtherStep"
          />
          <ProjectAdminFormstyle__StepCreateButton
            bsStyle="primary"
            id="js-btn-create-step"
            title={
              <React.Fragment>
                <i
                  className="fa fa-plus-circle"
                />
                 
                <FormattedMessage
                  id="global.add"
                  values={Object {}}
                />
              </React.Fragment>
            }
          >
            <ProjectAdminFormstyle__StepMenuItem
              id="presentation_step"
              key="0"
              onClick={[Function]}
            >
              <FormattedMessage
                id="presentation_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="collect_step"
              key="1"
              onClick={[Function]}
            >
              <FormattedMessage
                id="collect_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="selection_step"
              key="2"
              onClick={[Function]}
            >
              <FormattedMessage
                id="selection_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.consultation"
              key="3"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.consultation"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.synthesis"
              key="4"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.synthesis"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.ranking"
              key="5"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.ranking"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.questionnaire"
              key="6"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.questionnaire"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.custom.feminine"
              key="7"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.custom.feminine"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
          </ProjectAdminFormstyle__StepCreateButton>
        </ButtonToolbar>
      </div>
    </div>
  </ProjectAdminFormstyle__ProjectBoxContainer>
</div>
`;

exports[`<ProjectStepAdmin /> renders correctly without feature debate 1`] = `
<div
  className="col-md-12"
>
  <ProjectAdminFormstyle__ProjectBoxContainer
    className="box container-fluid"
  >
    <ProjectAdminFormstyle__ProjectBoxHeader>
      <h4>
        <FormattedMessage
          id="project.show.meta.step.title"
          values={Object {}}
        />
      </h4>
    </ProjectAdminFormstyle__ProjectBoxHeader>
    <div
      className="box-content"
    >
      <div
        className="form-group"
        id="project_form_admin_steps_panel"
      >
        <FieldArray
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "__ComponentClass": [Function],
              "displayName": "Relay(Connect(ProjectStepAdminList))",
              "render": [Function],
            }
          }
          formName="testForm"
          name="steps"
          project={
            Object {
              "$fragmentRefs": null,
              "$refType": null,
            }
          }
        />
        <ButtonToolbar
          bsClass="btn-toolbar"
        >
          <Relay(ProjectAdminStepFormModal)
            form="testForm"
            isCreating={true}
            onClose={[Function]}
            project={
              Object {
                "$fragmentRefs": null,
                "$refType": null,
              }
            }
            show={false}
            step={null}
            type="OtherStep"
          />
          <ProjectAdminFormstyle__StepCreateButton
            bsStyle="primary"
            id="js-btn-create-step"
            title={
              <React.Fragment>
                <i
                  className="fa fa-plus-circle"
                />
                 
                <FormattedMessage
                  id="global.add"
                  values={Object {}}
                />
              </React.Fragment>
            }
          >
            <ProjectAdminFormstyle__StepMenuItem
              id="presentation_step"
              key="0"
              onClick={[Function]}
            >
              <FormattedMessage
                id="presentation_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="collect_step"
              key="1"
              onClick={[Function]}
            >
              <FormattedMessage
                id="collect_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="selection_step"
              key="2"
              onClick={[Function]}
            >
              <FormattedMessage
                id="selection_step"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.debate"
              key="3"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.debate"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.consultation"
              key="4"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.consultation"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.synthesis"
              key="5"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.synthesis"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.ranking"
              key="6"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.ranking"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.questionnaire"
              key="7"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.questionnaire"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
            <ProjectAdminFormstyle__StepMenuItem
              id="global.custom.feminine"
              key="8"
              onClick={[Function]}
            >
              <FormattedMessage
                id="global.custom.feminine"
                values={Object {}}
              />
            </ProjectAdminFormstyle__StepMenuItem>
          </ProjectAdminFormstyle__StepCreateButton>
        </ButtonToolbar>
      </div>
    </div>
  </ProjectAdminFormstyle__ProjectBoxContainer>
</div>
`;
